splice_machine/src/main/java/com/splicemachine/derby/stream/control/ControlDataSet.java

```diff
@@ -496,6 +496,12 @@ public String getAttribute(String name) {
         throw new UnsupportedOperationException("Not Implemented in Control Side");
     }
 
+    @Override
+    public DataSet<V> crossJoin(OperationContext operationContext, DataSet<V> rightDataSet) {
+        throw new UnsupportedOperationException("Not Implemented in Control Side");
```

Comment:
I'd implement this and wouldn't check on CrossJoinOperation for Spark vs Control implementations.

Reply:
This is like `ControlDataset.join()`, which is not implemented. The cross join implementation for control is implemented in `CrossJoinOperation`, like other join operations. So I think it would be better to make it consistent.